addons/osfstorage/models.py

```diff
@@ -387,7 +387,7 @@ def delete(self, user=None, parent=None, **kwargs):
         from website.search import search
 
         search.update_file(self, delete=True)
-        return super(OsfStorageFile, self).delete(user, parent, **kwargs)
+        return super().delete(user, parent, **kwargs)
```

Comment:
`parent` must be removed here as well (and preferably from the function definition too) -- `super`'s call won't have it, so its default of `None` overrides `save=True` as the next positional argument in `BaseFileNode.delete`.

Also, not your `git blame`, but prefer passing keyword arguments with their keywords to avoid this type of issue.